src/elements/content-explorer/ContentExplorer.js

```diff
@@ -383,6 +386,40 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    shouldFetchThumbnails(items: ?Array<BoxItem>) {
+        const { viewMode } = this.state;
+        return viewMode === VIEW_MODE_GRID && items;
+    }
+
+    /**
+     * Call Box Representations API to fetch thumbnails of all files in collection.
+     * The thumbnailUrls array in state will be populated with urls to thumbnails.
+     * If a thumbnail could not be found, that item's entry in thumbnailUrls will be null.
+     *
+     * @private
+     * @param {Array<BoxItem>} items - item collection object
+     * @param {string} dimensions - desired dimensions of thumbnail. Acceptable dimensions
+     * for a jpg are: "32x32", "94x94", "160x160", "320x320", "1024x1024", "2048x2048".
+     * @return {void}
+     */
+    fetchThumbnailUrls(items: Array<BoxItem> = [], dimensions: string): void {
```

Comment:
Could we include the `representations` field in the initial folder fetch request?

Reply:
It seems like the first part of this method should live in either the `Folder` or `File` API helper.